lib/components/type_ahead.ex

```diff
@@ -12,12 +12,13 @@ defmodule CrunchBerry.Components.TypeAhead do
           label: String.t(),
           search_text: String.t(),
           search_results: [] | [{integer(), String.t()}],
-          current_focus: integer()
+          current_focus: integer(),
+          target: String.t()
```

Comment:
Not sure if Dialyzer will pick this up, but `@myself` is a `CID` struct

```elixir
optional(:target) => String.t() | %Phoenix.LiveComponent.CID{}
```

Reply:
Perhaps I am mistaken, but I thought it was returning a string that is an id selector? That was what it looked like when I was printing the params out.

Reply:
`CID` implements the `String.Chars` protocol, so when you print it you'll just see the ID it contains, but it's really a struct under the hood (see [the source](https://github.com/phoenixframework/phoenix_live_view/blob/v0.15.7/lib/phoenix_live_component.ex#L493)).

Reply:
Ahh ok. Thanks for this clarification!!

Reply:
I'm surprised the dialyzer didn't complain.